kolibri/plugins/learn/assets/src/views/ContentCardListViewItem.vue

```diff
@@ -182,13 +231,39 @@
       },
       kindToLearningActivity() {
         let activity = '';
-        if (Object.values(LearningActivities).includes(this.kind)) {
+        if (this.kind === 'topic') {
+          return 'folder';
+        } else if (Object.values(LearningActivities).includes(this.kind)) {
           activity = this.kind;
+          return `${activity}`;
         } else {
           // otherwise reassign the old content types to the new metadata
           activity = ContentKindsToLearningActivitiesMap[this.kind];
+          return `${activity}`;
         }
-        return activity;
+      },
+      iconLabel() {
+        const kindToLearningActivity = this.coreString(this.kindToLearningActivity);
+        let addedMobileDisplay = '';
+        // append the | and the activity length on mobile views only
+        // make all part of the same string for correct RTL display
+        if (this.isMobile) {
+          addedMobileDisplay = ` | ${this.coreString(this.activityLength)}`;
+        }
+        return `${kindToLearningActivity} ${addedMobileDisplay}`;
+      },
+      isLibraryPage() {
+        return this.pageName === PageNames.LIBRARY;
+      },
+      ceilingDate() {
+        if (this.createdDate > this.now) {
+          return this.now;
+        }
+        return this.createdDate;
+      },
+      bookmarkCreated() {
+        const time = this.$formatRelative(this.ceilingDate, { now: this.now });
```

Comment:
Will the time be correct if I have a page with this component open for a couple of minutes and then bookmark something? I haven't tested this myself yet though from what I see in code my guess would be that right now `this.now` would rather return time when this component was mounted and not necessarily when a bookmark was really created.

Reply:
This card component is used for the bookmarks page itself, and so I am trying to figure out a way where it might be mounted and then have a bookmark created after without a re-navigation. I have tried testing this and so far it seems that the bookmarks time is reflecting accurately. But if you have a specific scenario in mind (or if someone else has one!), I would be happy to give it a try!

Reply:
Thank you for checking it. From your explanation, it seems that it creates a hidden dependency from where this component is being used so even though things work for the bookmarks page, I'd recommend using something like [this removed implementation](https://github.com/learningequality/kolibri/pull/8461/files?authenticity_token=%2FOJGzRbCqz5GWJWPjneZGBH5FyeWWt%2BAvhmW9SXn9yzooCYichAcOdCfEN6t5btF%2BK1cQ72w9qRN22Fv%2FNwh8w%3D%3D&file-filters%5B%5D=.scss&file-filters%5B%5D=.vue#diff-96160db5aef4cbd2958957ee894db8c453a6b4dbfa372f557187af54855a1242L106-L113) so that we can be sure that it will work as expected no matter of scenario since this is a general component. Related to that, `ContentCardListViewItem` is used in `ContentCardGroupGrid` which seems to be imported into another four pages.

Reply:
Sure, and regarding those components being used from both non-hybrid-learning and hybrid-learning, I hope we can find something simple to not block this PR way too much :) Just from the code, those various dependencies look to be "meeeh" :)) Let me know please if I can help with that somehow.